cli/src/commands/log.rs

```diff
@@ -150,12 +153,33 @@ pub(crate) fn cmd_log(
         }
         expression
     };
+
+    let revset = revset_expression.evaluate()?;
+
+    if args.count {
+        let count = if args.limit.is_none() {
+            let (lower, upper) = revset.count_estimate()?;
+            if upper == Some(lower) {
+                lower
+            } else {
+                revset.iter().process_results(|iter| iter.count())?
+            }
+        } else {
+            revset
+                .iter()
+                .take(args.limit.unwrap_or(usize::MAX))
+                .process_results(|iter| iter.count())?
+        };
```

Comment:
nit: we can use the `count_estimate()` even if `--limit` is used:
```rust
        let (lower, upper) = revset.count_estimate()?;
        if upper == Some(lower) {
            min(lower, args.limit.unwrap_or(usize::MAX))
        } else {
            revset
                .iter()
                .take(args.limit.unwrap_or(usize::MAX))
                .process_results(|iter| iter.count())?
        }
```

(We can also avoid iterating if e.g. the estimate is "> 10" and the limit is 5, but we can fix that later if it's common request.)